impeller/renderer/backend/vulkan/device_buffer_vk.cc

```diff
@@ -40,9 +40,20 @@ bool DeviceBufferVK::OnCopyHostBuffer(const uint8_t* source,
   if (source) {
     ::memmove(dest + offset, source + source_range.offset, source_range.length);
   }
-  ::vmaFlushAllocation(resource_->buffer.get().allocator,
-                       resource_->buffer.get().allocation, offset,
-                       source_range.length);
+
+  vk::PhysicalDeviceMemoryProperties memory_properties;
+  device.getMemoryProperties(&memory_properties);
+
+  bool isHostCoherent =
```

Comment:
Rather than looking this up on each `OnCopyHostBuffer`, can we do this in allocator_vk and store the result?

There is also the DeviceBufferVK::Flush method below, which could be updated to no-op when this is true, but may be called more often.

Reply:
Sure, will check this